lib/vmpooler/pool_manager.rb

```diff
@@ -619,8 +619,24 @@ def load_used_providers
     # @return [Array] - a list of used providers from the config file, defaults to the default providers
     # ie. ["dummy"]
     def used_providers
+      # create an array of provider classes based on the config
+      if config[:providers]
+        config_provider_names = config[:providers].keys
+        config_providers = config_provider_names.map do |config_provider_name|
+          if config[:providers][config_provider_name] && config[:providers][config_provider_name]['provider_class']
+            config[:providers][config_provider_name]['provider_class']
+          else
+            config_provider_name
+          end
+        end.compact.uniq
+      else
+        config_providers = []
+      end
+      # create an array of providers specified by pools
       pools = config[:pools] || []
-      @used_providers ||= (pools.map { |pool| pool[:provider] || pool['provider'] }.compact + default_providers).uniq
+      pool_provider_names = pools.map { |pool| pool[:provider] || pool['provider'] }.compact.uniq
```

Comment:
not sure this list is useful. It would get provider names from the pools but these normally point to providers listed in the providers hash.

Reply:
You have a point. Let me see if any tests blow up without it

Reply:
Seems okay, pushed